programs/main.cpp

```diff
@@ -332,6 +334,19 @@ struct Checker
 #endif
 ;
 
+void checkHarmfulEnvironmentVariables()
+{
+    /// The list is a selection from "man ld-linux":
+    for (auto var : {"LD_PRELOAD", "LD_LIBRARY_PATH", "LD_ORIGIN_PATH", "LD_AUDIT", "LD_DYNAMIC_WEAK"})
```

Comment:
Worth adding `DYLD_INSERT_LIBRARIES` ?

Also maybe it worth doing this only for non-shared builds, since shared builds are for debugging only, and sometimes it is useful to replace libraries using dynamic linker.

Reply:
Yes. Where to read about all similar variables?

Reply:
I don't have any place, I just remember that osx linker has different variables.
Here is a list for osx - https://www.manpagez.com/man/1/dyld/

By the way, I completely forgot about `DYLD_LIBRARY_PATH` (this is one of generic variables)